custom/icds_reports/reports/service_delivery_dashboard_data.py

```diff
@@ -0,0 +1,352 @@
+            awc_name=get_value_or_data_not_entered(row_data, 'awc_name'),
+            num_launched_awcs=get_value_or_data_not_entered(row_data, 'num_launched_awcs')
+        )
+        if step == 'thr':
+            base_dict = get_thr_percents(base_dict, row_data)
+            base_dict['thr_0_days'] = get_value_or_data_not_entered(row_data, 'thr_0_days')
+            base_dict['thr_1_7_days'] = get_value_or_data_not_entered(row_data, 'thr_1_7_days')
+            base_dict['thr_8_14_days'] = get_value_or_data_not_entered(row_data, 'thr_8_14_days')
+            base_dict['thr_15_20_days'] = get_value_or_data_not_entered(row_data, 'thr_15_20_days')
+            base_dict['thr_21_24_days'] = get_value_or_data_not_entered(row_data, 'thr_21_24_days')
+            base_dict['thr_25_days'] = get_value_or_data_not_entered(row_data, 'thr_25_days')
+            base_dict['thr_eligible'] = get_value_or_data_not_entered(row_data, 'thr_eligible')
+        elif step == 'cbe':
+            base_dict['cbe_conducted'] = get_value_or_data_not_entered(row_data, 'cbe_conducted')
+            base_dict['third_fourth_month_of_pregnancy_count'] =\
+                get_value_or_data_not_entered(row_data, 'third_fourth_month_of_pregnancy_count')
+            base_dict['annaprasan_diwas_count'] = get_value_or_data_not_entered(row_data,
+                                                                                'annaprasan_diwas_count')
+            base_dict['suposhan_diwas_count'] = get_value_or_data_not_entered(row_data,
+                                                                              'suposhan_diwas_count')
+            base_dict['coming_of_age_count'] = get_value_or_data_not_entered(row_data,
+                                                                             'coming_of_age_count')
+            base_dict['public_health_message_count'] =\
+                get_value_or_data_not_entered(row_data, 'public_health_message_count')
+        elif step == 'sn':
+            base_dict = get_sn_percents(base_dict, row_data)
+            base_dict['lunch_0_days'] = get_value_or_data_not_entered(row_data, 'lunch_0_days')
+            base_dict['lunch_1_7_days'] = get_value_or_data_not_entered(row_data, 'lunch_1_7_days')
+            base_dict['lunch_8_14_days'] = get_value_or_data_not_entered(row_data, 'lunch_8_14_days')
+            base_dict['lunch_15_20_days'] = get_value_or_data_not_entered(row_data, 'lunch_15_20_days')
+            base_dict['lunch_21_24_days'] = get_value_or_data_not_entered(row_data, 'lunch_21_24_days')
+            base_dict['lunch_25_days'] = get_value_or_data_not_entered(row_data, 'lunch_25_days')
+            base_dict['pse_eligible'] = get_value_or_data_not_entered(row_data, 'pse_eligible')
+        elif step == 'pse':
+            base_dict = get_pse_percents(base_dict, row_data)
+            base_dict['pse_0_days'] = get_value_or_data_not_entered(row_data, 'pse_0_days')
```

Comment:
same comment here to make it generalized  as thr, pse and lunch all three have same fields just the starting string is different

Reply:
changed

Reply:
hey you could do the same change for the `get_value_or_data_not_entered` fields as well no?

Reply:
done